aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -105,6 +105,12 @@ public String token() {
     return properties().get(OAuth2Properties.TOKEN);
   }
 
+  /**
+   * A token supplier that takes precedence over {@link S3V4RestSignerClient#token()} if it's set.
+   */
+  @Nullable
+  public abstract Supplier<String> tokenSupplier();
```

Comment:
I feel like it's unnecessary to have both `token()` and `tokenSupplier()`.  I think we should just make `token()` be a supplier and then the default can always be `properties().get(OAuth2Properties.TOKEN`.  Having duplicates with precedence considerations just makes it more complicated.

Reply:
that makes a lot of sense. I've made those changes and pushed